src/components/Header/utils.ts

```diff
@@ -0,0 +1,265 @@
+      result.link = navLink
+    }
+  }
+
+  // Process dropdown items if they exist
+  if (tab.items && tab.items.length > 0) {
+    result.items = tab.items
+      .map((item) => {
+        if (!item) return null
+
+        const navItem: NavItem = {
+          id: item.id || generateRandomId(),
+        }
+
+        // Add link if it exists and is valid
+        if (item.link) {
+          const convertedLink = convertToNavLink(item.link)
+          if (convertedLink) {
+            navItem.link = convertedLink
+          }
+        }
+
+        // Process nested items if they exist
+        if (item.items && item.items.length > 0) {
+          navItem.items = item.items
+            .map((nestedItem) => {
+              if (!nestedItem) return null
+
+              const nestedNavItem: NavItem = {
+                id: nestedItem.id || generateRandomId(),
+              }
+
+              if (nestedItem.link) {
+                const convertedNestedLink = convertToNavLink(nestedItem.link)
+                if (convertedNestedLink) {
+                  nestedNavItem.link = convertedNestedLink
+                }
+              }
+
+              return nestedNavItem.link ? nestedNavItem : null
+            })
+            .filter(Boolean) as NavItem[]
+
+          // Only include items array if it has elements
+          if (navItem.items && navItem.items.length === 0) {
+            delete navItem.items
+          }
+        }
+
+        return navItem.link || (navItem.items && navItem.items.length > 0) ? navItem : null
+      })
+      .filter(Boolean) as NavItem[]
+
+    // Only include items if they exist
+    if (result.items && result.items.length === 0) {
+      delete result.items
+    }
+  }
+
+  // Only return the result if it has a link or items
+  return result.link || (result.items && result.items.length > 0) ? [result] : []
+}
+
+/**
+ * Helper function to convert a payload navLink (@/fields/navLinks) to a frontend NavLink
+ * Returns undefined if the link is invalid
+ */
+function convertToNavLink(link: {
+  type?: ('internal' | 'external') | null
+  reference?:
+    | ({
+        relationTo: 'pages'
+        value: number | Page
+      } | null)
+    | ({
+        relationTo: 'posts'
+        value: number | Post
+      } | null)
+  url?: string | null
+  label?: string | null
+  newTab?: boolean | null
+}): NavLink | undefined {
+  // Extract label from link or reference
+  let linkLabel: string | undefined = link.label || undefined
+
+  // Handle external links
+  if (link.type === 'external' && link.url) {
+    if (!linkLabel) {
+      return undefined // External links must have a label
+    }
+
+    return {
+      type: 'external',
+      label: linkLabel,
+      url: link.url,
+      newTab: !!link.newTab,
+    }
+  }
+
+  // Handle internal links with references
+  if (
+    link.type === 'internal' &&
+    link.reference &&
+    (link.reference.relationTo === 'pages' || link.reference.relationTo === 'posts')
```

Comment:
n.b. you could also formulate the output type as simply having a relative path and using it as an opaque value, in this function you know if it's a relation to pages or posts and can format the full relative path using the slug, then simplify the output type significantly to not need a `reference` sub-object, just a path.